lib/web/fetch/index.js

```diff
@@ -133,116 +178,147 @@ function fetch (input, init = undefined) {
 
-  // 6. If globalObject is a ServiceWorkerGlobalScope object, then set
-  // request’s service-workers mode to "none".
-  if (globalObject?.constructor?.name === 'ServiceWorkerGlobalScope') {
-    request.serviceWorkers = 'none'
-  }
+      // 2. Return p.
+      if (hasSubscribers) {
+        return handlePromise(p, context)
+      }
+      return p.promise
+    }
 
-  // 7. Let responseObject be null.
-  let responseObject = null
+    // 5. Let globalObject be request’s client’s global object.
+    const globalObject = request.client.globalObject
 
-  // 8. Let relevantRealm be this’s relevant Realm.
-  const relevantRealm = null
+    // 6. If globalObject is a ServiceWorkerGlobalScope object, then set
+    // request’s service-workers mode to "none".
+    if (globalObject?.constructor?.name === 'ServiceWorkerGlobalScope') {
+      request.serviceWorkers = 'none'
+    }
 
-  // 9. Let locallyAborted be false.
-  let locallyAborted = false
+    // 7. Let responseObject be null.
+    let responseObject = null
 
-  // 10. Let controller be null.
-  let controller = null
+    // 8. Let relevantRealm be this’s relevant Realm.
+    const relevantRealm = null
 
-  // 11. Add the following abort steps to requestObject’s signal:
-  addAbortListener(
-    requestObject.signal,
-    () => {
-      // 1. Set locallyAborted to true.
-      locallyAborted = true
+    // 9. Let locallyAborted be false.
+    let locallyAborted = false
 
-      // 2. Assert: controller is non-null.
-      assert(controller != null)
+    // 10. Let controller be null.
+    let controller = null
 
-      // 3. Abort controller with requestObject’s signal’s abort reason.
-      controller.abort(requestObject.signal.reason)
+    // 11. Add the following abort steps to requestObject’s signal:
+    addAbortListener(
+      requestObject.signal,
+      () => {
+        // 1. Set locallyAborted to true.
+        locallyAborted = true
 
-      // 4. Abort the fetch() call with p, request, responseObject,
-      //    and requestObject’s signal’s abort reason.
-      abortFetch(p, request, responseObject, requestObject.signal.reason)
-    }
-  )
+        // 2. Assert: controller is non-null.
+        assert(controller != null)
 
-  // 12. Let handleFetchDone given response response be to finalize and
-  // report timing with response, globalObject, and "fetch".
-  const handleFetchDone = (response) =>
-    finalizeAndReportTiming(response, 'fetch')
+        // 3. Abort controller with requestObject’s signal’s abort reason.
+        controller.abort(requestObject.signal.reason)
 
-  // 13. Set controller to the result of calling fetch given request,
-  // with processResponseEndOfBody set to handleFetchDone, and processResponse
-  // given response being these substeps:
+        // 4. Abort the fetch() call with p, request, responseObject,
+        //    and requestObject’s signal’s abort reason.
+        abortFetch(p, request, responseObject, requestObject.signal.reason, context)
```

Comment:
You're not using `context` in `abortFetch` anymore so you can remove this change.
```suggestion
        abortFetch(p, request, responseObject, requestObject.signal.reason)
```